src/db-connection.ts

```diff
@@ -71,7 +71,7 @@ export async function createDbConnection(dbConfig: Db = config.db): Promise<Knex
 
 export async function createReplicaDbConnection(
   replica_db_config: Replicadb = config.replica_db
-): Promise<Knex> {
+): Promise<{ connection: Knex; type: string }> {
```

Comment:
@dav1do this change makes the readReplica optional and by introducing a connectionType we have captured the two modes it can work in. 
The type is also useful to avoid misleading logs in the connection failure mode giving devs more insight into what is happening in the background